dbfit-java/derby/src/main/java/dbfit/environment/DerbyEnvironment.java

```diff
@@ -51,42 +53,153 @@ public Pattern getParameterPattern() {
             }
-            rs.close();
-            return allParams;
         }
     }
 
-    public Map<String, DbParameterAccessor> getAllProcedureParameters(
-            String procName) throws SQLException {
-        throw new UnsupportedOperationException();
+    private Map<String, DbParameterAccessor> createParamMap(ResultSet rs)
+            throws SQLException {
+        Map<String, DbParameterAccessor> allParams = new HashMap<>();
+        while (rs.next()) {
+            String paramName = defaultIfNull(rs.getString("COLUMN_NAME"), "");
+            String paramTypeName = rs.getString("TYPE_NAME");
+            allParams.put(NameNormaliser.normaliseName(paramName),
+                createDbParameterAccessor(
+                    paramName, getParamDirection(rs.getString("COLUMN_DIRECTION")),
+                    typeMapper.getJDBCSQLTypeForDBType(paramTypeName),
+                    getJavaClass(paramTypeName), rs.getInt("ORDINAL_POSITION") - 1));
+        }
+        return allParams;
+    }
+
+    private Direction getParamDirection(String direction) throws SQLException {
+        switch (direction) {
+            case "INPUT":
+                return Direction.INPUT;
+            case "INPUT_OUTPUT":
+                return Direction.INPUT_OUTPUT;
+            case "OUTPUT":
+                return Direction.OUTPUT;
+            case "RETURN_VALUE":
+                return Direction.RETURN_VALUE;
+            default:
+                throw new SQLException("Invalid column/parameter direction string " + direction);
+        }
+    }
+
+    private static class DatabaseObjectName {
+        String schemaName, objectName;
+
+        private DatabaseObjectName(String objSchemaName, String objName) {
+            schemaName = objSchemaName;
+            objectName = objName;
+        }
+
+        private String getSchemaName() {
+            return schemaName;
+        }
+
+        String getObjectName() {
+            return objectName;
+        }
+    }
+
+    DatabaseObjectName buildDatabaseObjectName(String objName) throws SQLException {
+        String[] qualifiers = objName.toUpperCase().split("\\.");
+        String schemaName = (qualifiers.length == 2) ? qualifiers[0] : getConnection().getSchema();
+        String objectName = (qualifiers.length == 2) ? qualifiers[1] : qualifiers[0];
+        return new DatabaseObjectName(schemaName, objectName);
     }
 
+    @Override
+    public Map<String, DbParameterAccessor> getAllProcedureParameters(String procName)
+            throws SQLException {
+        DatabaseObjectName name = buildDatabaseObjectName(procName);
+        // https://github.com/apache/derby/blob/10.13/java/engine/org/apache/derby/impl/jdbc/metadata.properties
+        String sql =
+        "SELECT V.COLUMN_NAME" +
+        "           AS COLUMN_NAME" +
+        "     , CASE WHEN A.ALIASTYPE = 'P'" +
+        "            THEN CASE WHEN V.COLUMN_TYPE = " + DatabaseMetaData.procedureColumnIn +
```

Comment:
We don't need to decode to some JDBC Type ID here. We can directly have it as string: `INPUT`, `OUTPUT`, ....